File: pnia/src/main/java/io/github/jlmc/reactive/domain/services/prefixes/PrefixesInitializer.java
package io.github.jlmc.reactive.domain.services.prefixes;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.context.ApplicationListener;
import org.springframework.context.event.ContextRefreshedEvent;
import org.springframework.stereotype.Component;

@Component
public class PrefixesInitializer implements ApplicationListener<ContextRefreshedEvent> {

    private static final Logger LOGGER = LoggerFactory.getLogger(PrefixesInitializer.class);

    private final PrefixesService prefixesService;

    public PrefixesInitializer(PrefixesService prefixesService) {
        this.prefixesService = prefixesService;
    }

    @Override
    public void onApplicationEvent(ContextRefreshedEvent event) {
        try {

            LOGGER.info("Initializing 'prefixes' service...");

            prefixesService.loadPrefixes();
        } catch (Exception e) {
            LOGGER.info("Unexpected problem happens on 'prefixes' service initialization!");
            throw new IllegalStateException(e);
        }
    }
}

File: pnia-kotlin/src/main/kotlin/io/github/jlmc/pniakotlin/domain/services/prefixes/PrefixesInitializer.kt
package io.github.jlmc.pniakotlin.domain.services.prefixes

import org.slf4j.LoggerFactory
import org.springframework.context.ApplicationListener
import org.springframework.context.event.ContextRefreshedEvent
import org.springframework.stereotype.Component

@Component
class PrefixesInitializer(
    private val prefixesService: PrefixesService
) :
    ApplicationListener<ContextRefreshedEvent?> {

    companion object {
        private val LOGGER = LoggerFactory.getLogger(PrefixesInitializer::class.java)
    }

    override fun onApplicationEvent(event: ContextRefreshedEvent) {
        try {
            LOGGER.info("Initializing 'prefixes' service...")
            prefixesService.loadPrefixes()
            LOGGER.info("Initialized 'prefixes' service!")
        } catch (e: Exception) {
            LOGGER.error("Unexpected problem happens on 'prefixes' service initialization!")
            throw IllegalStateException(e)
        }
    }
}
